amethyst_ui/src/pass.rs

```diff
@@ -36,18 +41,40 @@ struct CachedDrawOrder {
     pub cache: Vec<(f32, Entity)>,
 }
 
-/// Draw Ui elements, this uses target with name "amethyst_ui"
-/// `V` is `VertexFormat`
-#[derive(Clone, Debug)]
+/// A color used to query a hashmap for a cached texture of that color.
+struct KeyColor(pub [u8; 4]);
+
+impl Eq for KeyColor {}
+
+impl PartialEq for KeyColor {
+    fn eq(&self, other: &Self) -> bool {
+        self.0[0] == other.0[0] && self.0[1] == other.0[1] && self.0[2] == other.0[2]
+            && self.0[3] == other.0[3]
+    }
+}
+
+impl Hash for KeyColor {
+    fn hash<H: Hasher>(&self, hasher: &mut H) {
+        Hash::hash_slice(&self.0, hasher);
+    }
+}
+
+/// Draw Ui elements.  UI won't display without this.  It's recommended this be your last pass.
 pub struct DrawUi {
     mesh_handle: MeshHandle,
     cached_draw_order: CachedDrawOrder,
+    cached_color_textures: HashMap<KeyColor, TextureHandle>,
+    glyph_brushes: HashMap<
+        u32,
+        (
+            GlyphBrush<'static, Resources, Factory>,
+            WeakHandle<FontAsset>,
+        ),
+    >,
```

Comment:
Ugh, a typedef would probably be better here.

Reply:
It's 100% internal and this is the only place it's written out.  Would you still have me use a typedef for this?

Reply:
It just doesn't look very nice in code, that's all.